packages/components/src/components/carousel/carousel-styles.ts

```diff
@@ -183,24 +190,65 @@ export const getComponentCss = (
     //     display: block,
     //   }
     ...(hasPagination && {
-      pagination: {
+      ['pagination-container']: {
         ...buildResponsiveStyles(hasPagination, (hasPaginationValue: boolean) => ({
           display: hasPaginationValue ? 'flex' : 'none',
         })),
-        justifyContent: 'center',
+        position: 'relative',
+        justifyContent: isInfinitePagination ? 'flex-start' : 'center',
+        width: `calc(${paginationActiveBulletSize} + ${paginationBulletSize} * 4 + ${spacingStaticSmall} * 4)`, // Width for five bullets (one active + spacing)
+        left: 'calc(50% - 42px)',
+        overflowX: 'hidden',
+      },
+      pagination: {
+        display: 'flex',
+        alignItems: 'center',
+        width: 'fit-content',
         gap: spacingStaticSmall,
+        transition: `all ${carouselTransitionDuration}ms`,
+        ...(isInfinitePagination && {
+          [`& > .${bulletInfiniteClass} ~ span, & > .${bulletActiveClass} ~ span`]: {
+            width: paginationBulletSize,
+            height: paginationBulletSize,
+          },
+          [`& > .${bulletInfiniteClass} ~ .${bulletInfiniteClass} ~ span, & > .${bulletActiveClass} ~ .${bulletInfiniteClass} ~ span`]:
+            {
+              width: '0px',
+              height: '0px',
+            },
+          [`& > span.${bulletInfiniteClass}`]: {
+            width: paginationInfiniteBulletSize,
+            height: paginationInfiniteBulletSize,
+          },
+        }),
+        [`& > span.${bulletActiveClass}`]: {
+          background: isHighContrastMode ? canvasTextColor : primaryColor,
+          height: paginationBulletSize,
+          width: `${paginationActiveBulletSize} !important`,
+        },
       },
+      ...(isInfinitePagination && {
+        [`${paginationInfiniteClass}`]: {
+          ['& > .bullet:nth-child(-n+4)']: {
```

Comment:
Could resolve nesting by combining this selector directly on the previous line.

Reply:
JSS generates:
`.pagination--infinite\ \>\ \.bullet\:nth-child\(-n\+4\) {
  width: 8px;
  height: 8px;
}`